sd.go

```diff
@@ -0,0 +1,128 @@
+package gorillaz
+
+import (
+	"errors"
+	"fmt"
+	"github.com/hashicorp/consul/api"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/resolver"
+	"strings"
+	"time"
+)
+
+type AgentServiceRegistrationOpt func(sd *api.AgentServiceRegistration)
+
+type ServiceDefinition struct {
+	ServiceName                    string
+	Addr                           string
+	Port                           int
+	Tags                           []string
+	Meta                           map[string]string
+	Opt                            AgentServiceRegistrationOpt
+	DeregisterCriticalServiceAfter time.Duration
+	CheckInterval                  time.Duration
```

Comment:
Please add comments to what these times mean, same way as https://github.com/grpc/grpc-go/blob/master/keepalive/keepalive.go#L37

Reply:
EDIT: from what I see, we don't use it here.
Shouldn't these values go instead in the other piece of the puzzle, the etcd or consul part?

Reply:
I made a mistake here, I introduced a dependency on api.AgentServiceRegistration. 
The idea was instead to offer consul-agnostic options to configure service discovery